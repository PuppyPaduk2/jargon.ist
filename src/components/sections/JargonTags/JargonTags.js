// @flow

import React, { Fragment } from 'react';
import Link from 'gatsby-link';
import cx from 'classnames';

import { type Tag } from '../../../types/tag';

import './JargonTags.scss';

type Props = {
  tags: Array<Tag>,
  className?: string,
  max?: number,
};

const JargonTags = ({ tags, className, max = 6 }: Props) => (
  <div className={cx('c-jargon-tags', className)}>
    <Fragment>
      <ul className="c-jargon-tags__list">
        {tags.slice(0, max).map(tag => (
          <li key={tag.slug} className="c-jargon-tags__tag">
            <Link to={`/k/${tag.slug}`} className="c-jargon-tags__link">
              {tag.title}
            </Link>
          </li>
        ))}
      </ul>
      {tags.length > max && (
        <span className="c-jargon-tags__more">ve {tags.length - max} tane daha</span>
      )}
    </Fragment>
  </div>
);

JargonTags.defaultProps = {
  className: '',
  max: 6,
};

export default JargonTags;
