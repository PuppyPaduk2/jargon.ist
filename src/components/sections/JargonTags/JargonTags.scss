@import '../../../styles/abstracts/index';

.c-jargon-tags {
  &__tag {
    background-color: $color-comet;
    border-radius: px-to-rem(2px);
    box-shadow: 0 px-to-rem(2px) px-to-rem(4px) 0 rgba(0, 0, 0, 0.1);
    color: $color-white;
    font-size: $g-font-size-small;
    padding: pad(2xsmall);

    &:hover {
      background-color: darken($color-comet, 5);
      color: $color-white;
      text-decoration: none;
    }
  }

  &__more {
    color: $color-comet;
    font-size: $g-font-size-small;
    padding: pad(2xsmall);
  }
}
