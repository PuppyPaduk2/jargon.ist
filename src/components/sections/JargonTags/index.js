// @flow

import JargonTags from './JargonTags';

export default JargonTags;
