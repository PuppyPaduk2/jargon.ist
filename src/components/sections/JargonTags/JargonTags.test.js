// @flow

import React from 'react';
import { shallow } from 'enzyme';
import Link from 'gatsby-link';

import tagsFixture from '../../../../__fixtures__/tags';

import JargonTags from './JargonTags';

it('renders correctly', () => {
  const wrapper = shallow(<JargonTags tags={tagsFixture} />);

  expect(wrapper).toMatchSnapshot();
});

it('renders link to tags', () => {
  const wrapper = shallow(<JargonTags tags={tagsFixture} />);

  tagsFixture.forEach(tag => {
    expect(wrapper.find(Link).find(`[to='/k/${tag.slug}']`)).toHaveLength(1);
  });
});

it('slices the tags if exists more than max', () => {
  const tags = [
    { title: 'tag1', slug: 'tag1' },
    { title: 'tag2', slug: 'tag2' },
    { title: 'tag3', slug: 'tag3' },
    { title: 'tag4', slug: 'tag4' },
    { title: 'tag5', slug: 'tag5' },
    { title: 'tag6', slug: 'tag6' },
    { title: 'tag7', slug: 'tag7' },
    { title: 'tag8', slug: 'tag8' },
  ];

  const wrapper = shallow(<JargonTags tags={tags} max={3} />);

  expect(wrapper.text()).toContain('5 tane daha');
});
