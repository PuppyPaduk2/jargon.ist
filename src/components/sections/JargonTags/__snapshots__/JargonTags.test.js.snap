// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="c-jargon-tags"
>
  <React.Fragment>
    <ul
      className="c-jargon-tags__list"
    >
      <li
        className="c-jargon-tags__tag"
        key="tag1"
      >
        <GatsbyLink
          className="c-jargon-tags__link"
          to="/k/tag1"
        >
          tag1
        </GatsbyLink>
      </li>
      <li
        className="c-jargon-tags__tag"
        key="tag2"
      >
        <GatsbyLink
          className="c-jargon-tags__link"
          to="/k/tag2"
        >
          tag2
        </GatsbyLink>
      </li>
    </ul>
  </React.Fragment>
</div>
`;
